src/providers/wfs/qgswfsprovider.cpp

```diff
@@ -877,7 +877,17 @@ bool QgsWFSProvider::addFeatures( QgsFeatureList &flist, Flags flags )
       {
         the_geom.convertToMultiType();
       }
-      QDomElement gmlElem = QgsOgcUtils::geometryToGML( the_geom, transactionDoc );
+      QDomElement gmlElem;
+      // WFS 1.1.0 uses preferably GML 3, but ESRI mapserver in 2020 doesn't like it so we stick to GML2
+      /*
+      if ( mShared->mWFSVersion == QStringLiteral( "1.1.0" ) )
+      {
+        gmlElem = QgsOgcUtils::geometryToGML( the_geom, transactionDoc, QLatin1String( "GML3" ) );
```

Comment:
if this gets re-enabled, I guess we might have axis order issues. On a quick look the method signature you use for QgsOgcUtils::geometryToGML( ) will not take into account EPSG axis order, which will probably be a problem with compliant servers when using urn:ogc:def:crs:EPSG::4326. You would need the extended signature where the value of invertAxisOrientation would be true when crs.hasAxisInverted()

Shouldn't the settings on the reading part that mess with axis order (mURI.ignoreAxisOrientation() / mURI.invertAxisOrientation()) be applied there too ?  See QgsWFSSharedData::computeFilter() for the logic. Basically in WFS 1.0, we only take into account mURI.invertAxisOrientation()) (the user says "my server does broken things. invert the axis order"), and in WFS 1.1, we take into account both : if ignoreAxisOrientation() is set then it means "the server behaves as WFS 1.0 and it returns long, lat for EPSG:4326", otherwise EPSG:4326 will be "lat, lon" . And invertAxisOrientation() is applied on top of that.

Reply:
Sorry, I missed a commit in the cherry-pick, now it's the updated version. I agree on the full signature for geometryToGML, I'll try to apply the logic you recommend.

Reply:
@rouault I'm not sure I get it right: 

Is this the correct interpretation for WFS-T?

## WFS-T 1.1

```c++
const bool applyAxisInversion = ( crs().hasAxisInverted() && !  ignoreAxisOrientation() ) || invertAxisOrientation();
```

## WFS-T 1.0:

```c++
const bool applyAxisInversion = crs().hasAxisInverted() || invertAxisOrientation();
```

Reply:
@rouault I've tried to apply this logic in my last commit, I also refactored it into a method.

Reply:
I have tried to follow GeoServer conventions here: https://docs.geoserver.org/stable/en/user/services/wfs/axis_order.html

For QGIS Server WFS-T 1.1 that exposes srs as `EPSG:4326` I had to check for `useEPSGColumnFormat` and skip axis inversion in that case. 

This seems to work, I added a few QGIS Server integration tests for that. 

I'm not sure about Geomedia that was mentioned in a comment.

We should be close to a working solution, at least I hope so :smile_cat: